packages/grafana-ui/src/components/TimeSeries/utils.ts

```diff
@@ -197,6 +211,43 @@ export const preparePlotConfigBuilder: PrepConfig = ({ frame, theme, timeZone, g
       }
     }
   }
+
+  // Always publish events
+  builder.scaleKeys = [xScaleKey, yScaleKey];
+  const payload: DataHoverPayload = {
+    point: {
+      [xScaleKey]: null,
+      [yScaleKey]: null,
+    },
+    data: frame,
+  };
+  const hoverEvent = new DataHoverEvent(payload);
+  builder.setCursor({
+    sync: {
+      key: '__global_',
+      filters: {
+        pub: (type: string, src: uPlot, x: number, y: number, w: number, h: number, dataIdx: number) => {
+          payload.columnIndex = dataIdx;
+          if (x < 0 && y < 0) {
+            payload.point[xScaleKey] = null;
+            payload.point[yScaleKey] = null;
+            eventBus.publish(new DataHoverClearEvent(payload));
+          } else {
+            // convert the points
+            payload.point[xScaleKey] = src.posToVal(x, xScaleKey);
+            payload.point[yScaleKey] = src.posToVal(y, yScaleKey);
+            eventBus.publish(hoverEvent);
+            hoverEvent.payload.down = undefined;
+          }
+          return true; // false; // for now sync everything all the time
```

Comment:
@dprokop -- IIUC, returning true will let uplot internally handle the events, returning blocks it (ie not broadcast to others)

this means we are currently doing both internal and external!  Might be fine to just always handle explicitly